packages/esm-patient-registration-app/src/patient-registration/input/custom-input/identifier/utils.test.ts

```diff
@@ -0,0 +1,75 @@
+import { isUniqueIdentifierTypeForOffline, shouldBlockPatientIdentifierInOfflineMode } from './utils';
+
+function createIdentifierType(options: any = {}) {
+  return {
+    uniquenessBehavior: (options.uniquenessBehavior as 'UNIQUE' | 'LOCATION' | 'NON_UNIQUE') || null,
+    identifierSources: [
+      {
+        uuid: 'identifier-source-uuid',
+        name: 'Identifier Source Name',
+        autoGenerationOption: {
+          manualEntryEnabled: (options.manualEntryEnabled as boolean) || false,
+          automaticGenerationEnabled: (options.automaticGenerationEnabled as boolean) || true,
+        },
+      },
+    ],
+    name: 'Identifier Type Name',
+    required: true,
+    uuid: 'identifier-type-uuid',
+    fieldName: 'identifierFieldName',
+    format: 'identifierFormat',
+    isPrimary: true,
+  };
+}
+
+describe('shouldBlockPatientIdentifierInOfflineMode', () => {
+  it('should return false if identifierType is unique and no manual entry is enabled', () => {
+    const identifierType = createIdentifierType({ uniquenessBehavior: null });
+
+    const result = shouldBlockPatientIdentifierInOfflineMode(identifierType);
+
+    expect(result).toBe(false);
+  });
+
+  it('should return false if identifierType is not unique', () => {
+    const identifierType = createIdentifierType({ uniquenessBehavior: null });
+
+    const result = shouldBlockPatientIdentifierInOfflineMode(identifierType);
+
+    expect(result).toBe(false);
+  });
+
+  it('should return true if manual entry is enabled and identifierType is unique', () => {
```

Comment:
Let's have a format similar to the first test case 
```suggestion
  it('should return true if  identifierType is unique and manual entry is enabled', () => {
```